### ISSUE TITLE
Date is not validated correctly as type: string, format: date-time

### ISSUE THREAD

**Issue by severinwullschleger** [cdimascio/express-openapi-validator#246]
Hi

I want to return the _created_ and _lastModified_ timestamp from the database which I define in my ORM with type _Date_.

If I print the response object in my express router before returning it, it looks like this:
```
User {
   id: '1fe87c1d-5d31-4426-8757-aa5e62d2344e',
   lastModified: 2020-02-21T09:26:43.603Z,
   created: 2020-02-21T09:25:48.151Z,
```

If I call the endpoint I get the following reponse object:
```
{
  "data": [
    {
      "id": "1fe87c1d-5d31-4426-8757-aa5e62d2344e",
      "lastModified": "2020-02-21T09:26:43.603Z",
      "created": "2020-02-21T09:25:48.151Z"
    },
...
```
-> correctly stringified

As soon as I set the validateResponses option in the OpenApiValidator on true, I get the following error message:
```
{
  "error": ".response.data[0].lastModified should be string, .response.data[0].created should be string,
```
... although it is a string in the upper example

**Comment by martinbark** [comment]
I just hit this same issue today.  It's because the date is an object not a string.  At the point the response it validated it's still a Date object.  When you print it [date.toJSON()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON) is called to convert it to a string.  It's not specific to Date, you would see this issue on any object that has a [toJSON()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior) function

The only reliable way i found to work around this is to `JSON.stringify()` then `JSON.parse()` the response body before it's validated.  This means the response body is now formatted as the destination will see the message before it's validated.

I added this code after OpenApiValidator was been installed as a workaround
```
app.use((req, res, next) => {
  const original = res.json;
  res.json = function jsonp(_json) {
    return original.call(this, JSON.parse(JSON.stringify(json)));
  };
  next();
});
```

**Comment by severinwullschleger** [comment]
Thanks a lot. The workaround worked for me too.

**Comment by electrotype** [comment]
Using `send` seems to be required for me instead:

```
  app.use((req, res, next) => {
    const original = res.send;
    res.send = function jsonp(json) {
      return original.call(this, JSON.parse(JSON.stringify(json)));
    };
    next();
  });
```

**Comment by Dakuan** [comment]
horrible

**Comment by Dakuan** [comment]
> Using `send` seems to be required for me instead:
> 
> ```
>   app.use((req, res, next) => {
>     const original = res.send;
>     res.send = function jsonp(json) {
>       return original.call(this, JSON.parse(JSON.stringify(json)));
>     };
>     next();
>   });
> ```

unfortunately this will change your content type though...

**Comment by cdimascio** [comment]
Thanks @Dakuan, this being worked on. You can follow along here. https://github.com/cdimascio/express-openapi-validator/pull/493

**Comment by cdimascio** [comment]
This is resolved in `v4.10.0`. please give it a try!
